### PR TITLE
fix(compose): container locking

### PR DESCRIPTION
**Pull request by stevenh**
Fix container locking which was duplicated and held for the duration of the concurrent operation so preventing any concurrency.

### CODE REVIEW

File: modules/compose/compose_api.go [hunk 1]
```diff
@@ -474,15 +465,14 @@ func (d *dockerCompose) lookupContainer(ctx context.Context, svcName string) (*t
 
 	ctr.SetProvider(dockerProvider)
 
+	d.containersLock.Lock()
```
Comment (mdelapenya) [inline modules/compose/compose_api.go:468]: Totally right, makes it more sense to lock inside the lookup method. Good catch!

**Review by mdelapenya — APPROVED** [review]
LGTM, thanks!

**Review by mdelapenya — CHANGES_REQUESTED** [review]
BTW, could you take a look at the test errors? they seem narrowed to the compose module, so they seem legit

**Comment by stevenh** [comment]
> BTW, could you take a look at the test errors? they seem narrowed to the compose module, so they seem legit

Yep, there was other fixes in later versions which we're key for this one. I've back ported those, and will check the tests once they complete.

**Review by mdelapenya — APPROVED** [review]
LGTM, thanks!

**Pull request merged**